Volume/Views/DebriefArticleView.swift

```diff
@@ -0,0 +1,136 @@
+                .frame(width: 44, height: 44)
+                .overlay(Circle().stroke(Color.volume.orange, lineWidth: 4))
+                .clipShape(Circle())
+
+                Button {
+                    incrementShoutouts(for: article)
+                } label: {
+                    let _ = print("the shoutouts is \(max(article.shoutouts, userData.shoutoutsCache[article.id, default: 0]))")
+                    Image("shout-out")
+                        .resizable()
+                        .foregroundColor(max(article.shoutouts, userData.shoutoutsCache[article.id, default: 0]) > 0 ? Color.white : Color.volume.orange)
+                        .background(max(article.shoutouts, userData.shoutoutsCache[article.id, default: 0]) > 0 ? Color.volume.orange : Color.white)
+                        .scaledToFit()
+                        .frame(height: 21)
+                }
+                .frame(width: 44, height: 44)
+                .background(max(article.shoutouts, userData.shoutoutsCache[article.id, default: 0]) > 0 ? Color.volume.orange : Color.white)
+                .overlay(Circle().stroke(Color.volume.orange, lineWidth: 4))
+                .clipShape(Circle())
+            }
+            .padding([.leading, .trailing], 65)
+            .padding(.top, 30)
+            Spacer()
+        })
+    }
+
+    func displayShareScreen(for article: Article) {
+        let rawString = Secrets.openArticleUrl + article.id
+        if let shareArticleUrl = URL(string: rawString) {
+            let rootVC = UIApplication.shared.windows.first?.rootViewController
+            let presentingVC = (rootVC?.presentedViewController ?? rootVC)
+            let linkSource = LinkItemSource(url: shareArticleUrl, article: article)
+            let shareVC = UIActivityViewController(activityItems: [linkSource], applicationActivities: nil)
+            presentingVC?.present(shareVC, animated: true, completion: nil)
+        }
+    }
+
+    private func incrementShoutouts(for article: Article) {
+        guard let uuid = userData.uuid else { return }
+
+        AppDevAnalytics.log(VolumeEvent.shoutoutArticle.toEvent(.article, value: article.id, navigationSource: .weeklyDebrief))
+        userData.incrementShoutoutsCounter(article)
+        let currentArticleShoutouts = max(userData.shoutoutsCache[article.id, default: 0], article.shoutouts)
+        userData.shoutoutsCache[article.id, default: 0] = currentArticleShoutouts + 1
+        let currentPublicationShoutouts = max(userData.shoutoutsCache[article.publication.id, default: 0], article.publication.shoutouts)
+        userData.shoutoutsCache[article.publication.id, default: 0] = currentPublicationShoutouts + 1
+        cancellableShoutoutMutation = Network.shared.publisher(for: IncrementShoutoutsMutation(id: article.id, uuid: uuid))
+            .sink(receiveCompletion: { completion in
+                if case let .failure(error) = completion {
+                    print(error.localizedDescription)
```

Comment:
Maybe we should say where the error came from. We don't really give this much attention throughout the app; your call if we should. I think it's a really really minor point.